test/unit/content-negotiation.spec.ts

```diff
@@ -1,126 +1,233 @@
+    describe('exact matches', () => {
+        it('matches none', () => {
+            expect(findMatchingType('xxx', ['aaa', 'bbb'])).toBe(undefined);
+        });
+
+        it('matches one', () => {
+            expect(findMatchingType('aaa', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('bbb', ['aaa', 'bbb'])).toBe('bbb');
+        });
+
+        it('matches one of many', () => {
+            expect(findMatchingType('aaa,xxx', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('xxx,aaa', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('bbb,xxx', ['aaa', 'bbb'])).toBe('bbb');
+            expect(findMatchingType('xxx,bbb', ['aaa', 'bbb'])).toBe('bbb');
+        });
+
+        it('matches one of many, ignoring whitespaces', () => {
+            expect(findMatchingType('aaa, xxx', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('xxx, aaa', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('bbb, xxx', ['aaa', 'bbb'])).toBe('bbb');
+            expect(findMatchingType('xxx, bbb', ['aaa', 'bbb'])).toBe('bbb');
+        });
+    });
+
+    describe('with parameters', () => {
+        it('prefers exact matches, but falls back accordingly', () => {
+            expect(findMatchingType('aaa;v2', ['aaa', 'aaa;v2'])).toBe('aaa;v2');
+            expect(findMatchingType('aaa', ['aaa', 'aaa;v2'])).toBe('aaa');
+            expect(findMatchingType('aaa;v2', ['aaa'])).toBe('aaa');
+            expect(findMatchingType('aaa;v3', ['aaa;v1', 'aaa;v2'])).toBe('aaa;v1');
+        });
+
+        it('prefers exact matches, but falls back accordingly, ignoring whitespaces', () => {
+            expect(findMatchingType('aaa; v2', ['aaa', 'aaa;v2'])).toBe('aaa;v2');
+            expect(findMatchingType('aaa;v2', ['aaa', 'aaa; v2'])).toBe('aaa; v2');
+            expect(findMatchingType('aaa', ['aaa', 'aaa; v2'])).toBe('aaa');
+            expect(findMatchingType('aaa; v2', ['aaa'])).toBe('aaa');
+        });
+    });
+
+    describe('extensions', () => {
+        it('matches exactly', () => {
+            expect(findMatchingType('application/vnd.foo+json', ['application/json', 'application/vnd.foo+json'])).toBe(
+                'application/vnd.foo+json'
+            );
+            expect(findMatchingType('application/json', ['application/json', 'application/vnd.foo+json'])).toBe(
+                'application/json'
+            );
+        });
+
+        it('fallsback to base content-type', () => {
+            expect(findMatchingType('application/vnd.foo+json', ['application/json'])).toBe('application/json');
+        });
+
+        it('uses extended content-type', () => {
+            expect(findMatchingType('application/json', ['application/vnd.foo+json'])).toBe('application/vnd.foo+json');
+        });
+    });
+
+    describe('wildcard responses', () => {
+        it('matches wildcard types', () => {
+            expect(findMatchingType('aaa', ['aaa/*'])).toBe('aaa/*');
```

Comment:
Is this even a thing? What would it mean for an OAS to say I return a wild card subtype?

Reply:
I've seen examples of this, where the OAS mentions things beyond the type/subtype (eg. `application/json; charset=utf-8`). For completeness, I assume it makes sense to allow the OAS to specify `image/*` meaning the provider can return with any image type.

Reply:
👉 https://swagger.io/docs/specification/media-types/

<img width="1116" alt="Screenshot 2023-05-24 at 2 56 24 pm" src="https://github.com/pactflow/swagger-mock-validator/assets/53900/4abdc9b6-0556-4121-b63c-22229fce0710">

So, best you keep the test ;)